gpic/gravatar.go

```diff
@@ -108,35 +115,60 @@ func NewImage(inputs ...string) (*Image, error) {
 
 		i.disableDefault = false
 
-		if !isValidEmail(input) {
-			continue
-		}
+		if isValidEmail(input) {
+			emailHash, err := hashEmail(input)
+			if err != nil {
+				return nil, err
+			}
+			i.emailHash = emailHash
+			i.email = input
 
-		emailHash, err := hashEmail(input)
-		if err != nil {
-			return nil, err
-		}
-		i.emailHash = emailHash
-		i.email = input
+			err = i.SetSize(80)
+			if err != nil {
+				return nil, err
+			}
 
-		err = i.SetSize(80)
-		if err != nil {
-			return nil, err
+			i.rating = RatingG
+
+			if len(inputs) != idx + 1 {
+				i.disableDefault = true
+				validURL, err := i.checkURL() 
+				if err != nil {
+					return nil, err	
+				}
+
+				if validURL {
+					break 
+				}
+			} 
 		}
 
-		i.rating = RatingG
+		if strings.HasPrefix(input, "gh:") {
+			
+			ghUser := strings.TrimPrefix(input, "gh:")
+			url := "https://api.github.com/users/" + ghUser
 
-		if len(inputs) != idx + 1 {
-			i.disableDefault = true
-			validURL, err := i.checkURL() 
+			resp, err := http.Get(url)
 			if err != nil {
-				return nil, err	
+				return nil, err
 			}
 
-			if validURL {
-				break 
+			defer resp.Body.Close()
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return nil, err
+			}
+			
+			var githubUser GithubUser 
+			err = json.Unmarshal(body, &githubUser)
+			if err != nil {
+				return nil, err
 			}
-		} 
+
+			fmt.Println(githubUser.AvatarURL) //Need to figure out what to do with this
```

Comment:
We ultamitely give it to the user via the struct. This is the prime purpose of this module. In the end, the user should be able to get a URL to a profile picture they can use in their app to represent a user.